cmd/libsnap-confine-private/device-cgroup-support.c

```diff
@@ -284,6 +284,19 @@ static struct rlimit _sc_cgroup_v2_adjust_memlock_limit(void) {
     return old_limit;
 }
 
+static bool _sc_is_snap_cgroup(const char *group) {
+    /* make a copy as basename may modify its input */
+    char *copy SC_CLEANUP(sc_cleanup_string) = sc_strdup(group);
```

Comment:
nitpick; I would rather allocate it as `char copy[PATH_MAX]`, to save a memory allocation.

Reply:
tweaked to use a buffer on stack and strncpy